code/game/mecha/combat/durand.dm

```diff
@@ -1,17 +1,209 @@
+/obj/mecha/combat/durand/proc/defense_check(var/turf/aloc)
+	if (!defense_mode || !shield || shield.switching)
+		return FALSE
+	. = FALSE
+	switch(dir)
+		if (1)
+			if(abs(x - aloc.x) <= (y - aloc.y) * -2)
+				. = TRUE
+		if (2)
+			if(abs(x - aloc.x) <= (y - aloc.y) * 2)
+				. = TRUE
+		if (4)
+			if(abs(y - aloc.y) <= (x - aloc.x) * -2)
+				. = TRUE
+		if (8)
+			if(abs(y - aloc.y) <= (x - aloc.x) * 2)
+				. = TRUE
+	return
+
+obj/mecha/combat/durand/attack_generic(mob/user, damage_amount = 0, damage_type = BRUTE, damage_flag = 0, sound_effect = 1, armor_penetration = 0)
+	if(defense_check(user.loc))
+//		log_message("Attack absorbed by defense field. Attacker - [user].", LOG_MECHA, color="orange")
+		shield.attack_generic(user, damage_amount, damage_type, damage_flag, sound_effect, armor_penetration)
+	else
+		. = ..()
+
+/obj/mecha/combat/durand/blob_act(obj/structure/blob/B)
+	if(defense_check(B.loc))
+//		log_message("Attack by blob. Attacker - [B].", LOG_MECHA, color="red")
+//		log_message("Attack absorbed by defense field.", LOG_MECHA, color="orange")
+		shield.blob_act(B)
+	else
+		. = ..()
+
+/obj/mecha/combat/durand/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(defense_check(user.loc))
+//		log_message("Attack absorbed by defense field. Attacker - [user], with [W]", LOG_MECHA, color="orange")
+		shield.attackby(W, user, params)
+	else
+		. = ..()
+
+/obj/mecha/combat/durand/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
+	if(defense_check(AM.loc))
+//		log_message("Impact with [AM] absorbed by defense field.", LOG_MECHA, color="orange")
+		shield.hitby(AM, skipcatch, hitpush, blocked, throwingdatum)
+	else
+		. = ..()
+
+////////////////////////////
+///// Shield processing ////
+////////////////////////////
+
+/**An object to take the hit for us when using the Durand's defense mode.
+It is spawned in during the durand's initilization, and always stays on the same tile.
+Normally invisible, until defense mode is actvated. When the durand detects an attack that should be blocked, the
+attack is passed to the shield. The shield takes the damage, uses it to calculate charge cost, and then sets its
+own integrity back to max. Shield is automatically dropped if we run out of power or the user gets out.*/
+
+/obj/durand_shield //projectiles get passed to this when defense mode is enabled
+	name = "defense grid"
+	icon = 'icons/mecha/durand_shield.dmi'
+	icon_state = "shield_null"
+	invisibility = INVISIBILITY_MAXIMUM //no showing on right-click
+	pixel_y = 4
+	max_integrity = 10000
+	obj_integrity = 10000
+	var/obj/mecha/combat/durand/chassis ///Our link back to the durand
+	var/switching = FALSE ///To keep track of things during the animation
+	anchored = TRUE
+
+/obj/durand_shield/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_MECHA_ACTION_ACTIVATE, .proc/activate)
+
+/obj/durand_shield/Destroy()
+	if(chassis)
+		chassis.shield = null
```

Comment:
add chassis = null

Reply:
Sorry, what's add chassis supposed to do? I don't know a whole lot and add. or add throws an error.

Reply:
ah, my bad. add this specific line without the two "
"chassis = null"

Reply:
> ah, my bad. add this specific line without the two " "chassis = null"

So.. I changed it to that and afterward it'd runtime whenever the shield was destroyed and get 'stuck' so it couldn't be turned on/off again.

Reply:
> > ah, my bad. add this specific line without the two " "chassis = null"
> 
> So.. I changed it to that and afterward it'd runtime whenever the shield was destroyed and get 'stuck' so it couldn't be turned on/off again. Wait, is it supposed to be l 145 or 146?

Add it after line 146. 
So, 
"/obj/durand_shield/Destroy()
	if(chassis)
		chassis.shield = null
                chassis = null
"
*rest of the code*

Reply:
Think I got it, thanks. Though it does cause a runtime when admin induced things happen (e.g. lowering the shield's HP to 1 and hitting it.)